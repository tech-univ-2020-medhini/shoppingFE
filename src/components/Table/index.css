table, th, td{
   border-collapse:collapse;
   padding: 1rem; 
}
.table-category{
    background-color: rgb(196,196,196);
    text-align: left;
}
.table{
    box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
}