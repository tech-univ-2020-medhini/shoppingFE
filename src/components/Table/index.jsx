import React from 'react';
import PropTypes from 'prop-types';
import './index.css';

function Table(props) {
  const { filteredProducts } = props;
  const productsJSX = [];
  if (filteredProducts !== undefined) {
    const categories = Object.keys(filteredProducts);
    categories.forEach((category) => {
      productsJSX.push(<tr><td className="table-category" colSpan={4}>{category}</td></tr>);
      const products = filteredProducts[category];
      products.forEach((product) => {
        productsJSX.push(
          <tr>
            <td>{product.name}</td>
            <td>{product.price}</td>
            <td>{product.cart}</td>
            <td>{product.price * product.cart}</td>
          </tr>,
        );
      });
    });
  }

  return (
    <div className="table">
      <table border={1}>
        <tr>
          <th>
            Item
          </th>
          <th>
            Price
          </th>
          <th>
            Quantity
          </th>
          <th>
            Subtotal
          </th>

        </tr>
        <tbody>
          {productsJSX}
        </tbody>

      </table>
    </div>

  );
}
Table.propTypes = {
  filteredProducts: PropTypes.objectOf(PropTypes.string).isRequired,
};

export default Table;
