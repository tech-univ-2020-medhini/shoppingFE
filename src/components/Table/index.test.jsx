import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Table from './index';


describe('the cart table', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Table />);

    expect(asFragment()).toMatchSnapshot();
  });
});
