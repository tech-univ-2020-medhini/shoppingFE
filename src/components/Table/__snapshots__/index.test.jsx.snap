// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the cart table should render correctly 1`] = `
<DocumentFragment>
  <div
    class="table"
  >
    <table
      border="1"
    >
      <tbody>
        <tr>
          <th>
            Item
          </th>
          <th>
            Price
          </th>
          <th>
            Quantity
          </th>
          <th>
            Subtotal
          </th>
        </tr>
      </tbody>
      <tbody />
    </table>
  </div>
</DocumentFragment>
`;
